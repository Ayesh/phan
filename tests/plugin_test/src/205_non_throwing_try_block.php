<?php
function examples205(int $flag, bool $x): bool {
    try {
        if ($x) {
            return true;
        }
    } catch (Exception $_) {
    }
    try {
        return $flag > 0;
    } catch (Exception $_) {
        echo "done\n";
    }
    for ($i = 0; $i < 10; $i++) {
        try {
            if ($i > 5) {
                echo "Found\n";
                break;
            }
        } catch (Exception $_) {
        }
    }
    return false;
}
examples205(1, false);
