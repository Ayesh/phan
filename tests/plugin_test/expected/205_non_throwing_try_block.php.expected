src/205_non_throwing_try_block.php:3 PhanPluginEmptyStatementPossiblyNonThrowingTryBody Found a try block that looks like it might not throw. Note that this check is a heuristic prone to false positives, especially because error handlers, signal handlers, destructors, and other things may all lead to throwing.
src/205_non_throwing_try_block.php:9 PhanPluginEmptyStatementPossiblyNonThrowingTryBody Found a try block that looks like it might not throw. Note that this check is a heuristic prone to false positives, especially because error handlers, signal handlers, destructors, and other things may all lead to throwing.
src/205_non_throwing_try_block.php:15 PhanPluginEmptyStatementPossiblyNonThrowingTryBody Found a try block that looks like it might not throw. Note that this check is a heuristic prone to false positives, especially because error handlers, signal handlers, destructors, and other things may all lead to throwing.
